docs/api/api.yml

```diff
@@ -386,38 +386,45 @@ paths:
     post:
       tags:
       - directory
-      summary: Adds a file or a subdirectory.
+      summary: Adds one or more files.
       description: |
-        Sub request of a request with `Content-Type` value `multipart/form-data`.
-        By default the files listed in the body of the main request are saved in `dir_path` under the file name provided
-        in the `Content-Disposition` header. However, if a file has the extension ZIP, it
-        is extracted in `dir_path` and `Content-Disposition` is ignored (and may be left out).
-        If `dir_path`, or any of its parents do not exist yet, they are first created.
+        The `Content-Type` must always be `multipart/form-data`, which means the body is formatted using the
+        [multipart media type](https://tools.ietf.org/html/rfc2046.html#section-5.1); each part contains one file.
+        The header area of each part specifies the file name using a `Content-Disposition` header.
+
+        If the body contains only one part and that part has `Content-Type: application/zip`, then the body
+        is a file in ZIP format. This ZIP file can contain a directory hierarchy. The ZIP file will be
+        unzipped in `dir_path`. For each file in the ZIP file, the path in the deposit will be `dir_path`/`path_in_zip`.
```

Comment:
What to do if the n-th file appears to be a ZIP? As the list of files is provided via an iterator, it is not logical to examine them all before processing them.

Reply:
The way it is specified now, the operation must then fail.

Reply:
should the files that were uploaded so far be deleted, or should all files be uploaded, except for the erroneous zipfile?

Reply:
So no second thought then ;)